src/com/owncloud/android/lib/common/operations/RemoteOperation.java

```diff
@@ -254,6 +254,34 @@ public Thread execute(OwnCloudClient client,
 		runnerThread.start();
 		return runnerThread;
 	}
+
+    /**
+     * Asynchronously executes the remote operation without a handler
+     *
+     * @param client			Client object to reach an ownCloud server
+     *                          during the execution of the operation.
+     * @param listener			Listener to be notified about the execution of the operation.
+     * @return					Thread were the remote operation is executed.
+     */
+    public Thread execute(OwnCloudClient client,
```

Comment:
Or even better, why should RemoteOperation care what it is calling back to? It should just callback to listener, its the listener responsibility to correct handle callback. Bu that depends on numbers of usages ;)